### PR TITLE
chore: extend test runner to accept transformers

### PR DESCRIPTION
**Pull request by KFlash**
@aladdin-add Can you check if this would work? See my TODO

### CODE REVIEW

**Comment by KFlash** [comment]
@aladdin-add Why the tests fail?

File: test/runner/tob.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { printSourceFile } from '../../src/printer';
-import { parseScript, parseModule } from '../../src/kataw';
+import { parseScript, parseModule, transformScript, transformModule } from '../../src/kataw';
```
Comment (aladdin-add) [inline test/runner/tob.ts:2]: seems `transformScript, transformModule` was not found. did you forget to push?

File: test/runner/tob.ts [hunk 2]
```diff
@@ -33,13 +36,20 @@ export async function file2Tob(filename: string): Promise<Tob> {
     filename,
     content,
     input: md2input(content),
+    transformOptions: md2transformOptions(content),
     parserOptions: md2parserOptions(content),
     printerOptions: md2printerOptions(content),
+    transform: md2cst(content),
```
Comment (aladdin-add) [inline test/runner/tob.ts:42]: should be md2transform??

the functions(md2xxx) could be simplified to one function.

File: test/runner/tob.ts (hunk 1 above)
Comment (KFlash) [reply test/runner/tob.ts:2]: Look in the main branch. They are there, so maybe sync the brances?

File: test/runner/tob.ts (hunk 2 above)
Comment (KFlash) [reply test/runner/tob.ts:42]: Can you fix that? Get this working?

File: test/runner/tob.ts (hunk 1 above)
Comment (KFlash) [reply test/runner/tob.ts:2]: https://github.com/kataw/kataw/blob/main/src/kataw.ts#L198-L204

File: test/__snapshot__/transformers/es2016/abstract.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
+# Kataw parser test case
+
+## Options
+
+`````js
+{ transform: true, transformers: [es2016], jsx: true, disableWebCompat: true, module: true }
```
Comment (KFlash) [inline test/__snapshot__/transformers/es2016/abstract.md:6]: @aladdin-add This test is what I need to get working. See the options

**Comment by KFlash** [comment]
Close this?

**Pull request closed without merging**